### ISSUE TITLE
saved model size

### ISSUE THREAD

**Issue by monsterlyg** [ultralytics/yolov5#3830]
## ❔Question
is the trained model saved with Fp16 ? but i find such message in class ModelEMA  in torch.utils.py
# Create EMA
        self.ema = deepcopy(model.module if is_parallel(model) else model).eval()  # FP32 EMA
However, when i save the model as FP32 through .float() explicitly, the size is about twice than the model saved by train.py.

**Comment by glenn-jocher** [comment]
@monsterlyg YOLOv5 🚀 training checkpoints should be 4X the size of final trained checkpoints, as they carry not just a FP16 model, but a FP16 [EMA](https://www.tensorflow.org/api_docs/python/tf/train/ExponentialMovingAverage) and an FP32 optimizer of the same size as the model (each model parameter has its own FP32 gradient saved within the optimizer). Final checkpoints contain only an FP16 model, with the EMA and optimizer both stripped after the final epoch of training:
https://github.com/ultralytics/yolov5/blob/75feeb797c4a9553f4274860c6c278f1fc628f60/train.py#L435-L439

The `strip_optimizer()` function updates the checkpoint dictionary value for `model`, replacing it with `ema`, and sets `ema` and `optimizer` keys to `None`, which will reduce the checkpoint size by 3/4.
https://github.com/ultralytics/yolov5/blob/75feeb797c4a9553f4274860c6c278f1fc628f60/utils/general.py#L495-L509

Final trained checkpoints after `strip_optimizer()` should match the README table sizes below.
<p align="center"><img width="700" alt="YOLOv5 Models" src="https://github.com/ultralytics/yolov5/releases/download/v1.0/model_comparison.png"></p>
